### PR TITLE
[#168752335] form fields validation

### PR DESCRIPTION
**Pull request by d-al-ibm**
added library react form hook
added strings for errors
added hook and form management in AddMailModal, RegistrationStepOne and RegistrationStepTwo

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Please include a Pivotal story at the beginning of the PR title (see below).</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>This PR changes a total of 586 LOCs, that is more than a reasonable size of 250. Consider splitting the pull request into smaller ones.</td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    

New dependencies added: <a href='https://www.npmjs.com/package/react-hook-form'><code>react-hook-form</code></a>.


<h2><a href="https://www.react-hook-form.com">react-hook-form</a></h2>
<p>Author: Unknown</p>
<p>Description: Performant, flexible and extensible forms library for React Hooks</p>
<p>Homepage: <a href="https://www.react-hook-form.com">https://www.react-hook-form.com</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>9 months ago</td></tr><tr><td>Last Updated</td><td>about 5 hours ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>470</td></tr><tr><td>Keywords</td><td>react, hooks, form, forms, form-validation, validation, typescript and react-hooks</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

<div align="center">
    <p align="center">
        <a href="https://react-hook-form.com" title="React Hook Form - Simple React forms validation">
            React Hook Form Logo - React hook custom hook for form validation
        </a>
    </p>
</div>

<p align="center">Performant, flexible and extensible forms with easy to use validation.</p>

<div align="center">

[CircleCI](https://circleci.com/gh/react-hook-form/react-hook-form)
[npm downloads](https://www.npmjs.com/package/react-hook-form)
[npm](https://www.npmjs.com/package/react-hook-form)
dep
[npm](https://badgen.net/bundlephobia/minzip/react-hook-form)
[Coverage Status](https://coveralls.io/github/bluebill1049/react-hook-form?branch=master)

[Tweet](https://twitter.com/intent/tweet?text=React+hooks+for+form+validation+without+the+hassle&url=https://github.com/bluebill1049/react-hook-form)&nbsp;[Join the community on Spectrum](https://spectrum.chat/react-hook-form)

</div>

<div align="center">
    <p align="center">
        <a href="https://react-hook-form.com" title="React Hook Form - Simple React forms validation">
            React Hook Form video - React custom hook for form validation
        </a>
    </p>
</div>

🇦🇺English | <a href="./docs/README.zh-CN.md">🇨🇳 简体中文</a> | <a href="./docs/README.ja-JP.md">🇯🇵 日本語</a> | <a href="./docs/README.ko-KR.md">🇰🇷한국어</a> | <a href="./docs/README.fr-FR.md">🇫🇷Français</a> | <a href="./docs/README.it-IT.md">🇮🇹Italiano</a> | <a href="./docs/README.pt-BR.md">🇧🇷Português</a>

## Features

- Built with performance and DX in mind
- Uncontrolled form validation
- [Tiny size](https://bundlephobia.com/result?p=react-hook-form@latest) without any dependency
- Follows HTML standard for validation
- Compatible with React Native
- Supports [Yup](https://github.com/jquense/yup) schema-based validation
- Supports native browser validation
- Build forms quickly with the [form builder](https://react-hook-form.com/form-builder)

## Install

    $ npm install react-hook-form

## Links

- [Motivation](https://medium.com/@bruce1049/form-validation-with-hook-in-3kb-c5414edf7d64)
- [Video tutorial](https://www.youtube.com/watch?v=-mFXqOaqgZk&t)
- [Get started](https://react-hook-form.com/get-started)
- [API](https://react-hook-form.com/api)
- [Examples](https://github.com/bluebill1049/react-hook-form/tree/master/examples)
- [Demo](https://react-hook-form.com)
- [Form Builder](https://react-hook-form.com/form-builder)
- [FAQs](https://react-hook-form.com/faqs)

## Quickstart

```jsx
import React from 'react';
import useForm from 'react-hook-form';

function App() {
  const { register, handleSubmit, errors } = useForm(); // initialise the hook
  const onSubmit = data => {
    console.log(data);
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <input name="firstname" ref={register} /> {/* register an input */}
      <input name="lastname" ref={register({ required: true })} />
      {errors.lastname && 'Last name is required.'}
      <input name="age" ref={register({ pattern: /\d+/ })} />
      {errors.age && 'Please enter number for age.'}
      <input type="submit" />
    </form>
  );
}
```

## Contributors

Thanks goes to these wonderful people. [[Become a contributor](CONTRIBUTING.md)].

<a href="https://github.com/react-hook-form/react-hook-form/graphs/contributors">
    
</a>

## Backers

Thanks goes to all our backers! [[Become a backer](https://opencollective.com/react-hook-form#backer)].

<a href="https://opencollective.com/react-hook-form#backers">
    
</a>


</details>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a54f6fb78b16e8dd1f9088873768b64a74ee9ae7
</p>

File: src/components/Modal/AddMailModal.tsx [hunk 1]
```diff
@@ -45,31 +52,24 @@ export const AddMailModal = withRouter((props: IAddMailModalProps) => {
...
+    IChangeMailFormData
+  >();
 
   /*
    * Boolean variable to check if a work email different from spid one is already set
    * */
-  const isWorkMailNotSet = !props.workMail || props.workMail === props.spidMail;
+  const isWorkMailNotSet =
+    (!props.workMail || props.workMail === props.spidMail) &&
+    props.location.pathname !== "/profile";
```
Comment (gunzip) [inline src/components/Modal/AddMailModal.tsx]: why we need the condition `props.location.pathname !== "/profile";` to check if the work email is not set ?

File: src/components/Registration/RegistrationStepOne/RegistrationStepOne.tsx [hunk 2]
```diff
@@ -168,32 +213,76 @@ export const RegistrationStepOne = withRouter(
...
+                                <RegistrationStepOneRadioButtons
+                                  name="organizationScope"
+                                  disabled={
+                                    props.isAdministrationAlreadyRegistered
+                                  }
+                                  dataArray={organizationScopes}
+                                  defaultCheckedValue={
+                                    props.selectedAdministration.scope
+                                  }
+                                  errors={errors}
```
Comment (gunzip) [inline src/components/Registration/RegistrationStepOne/RegistrationStepOne.tsx]: why do we need to pass these value here ? can't we just use `useForm` inside `RegistrationStepOneRadioButtons `?

File: src/components/Modal/AddMailModal.tsx (hunk 1 above)
Comment (d-al-ibm) [reply src/components/Modal/AddMailModal.tsx]: it was actually wrong, it is an additional condition to check to change modal/button texts based on location from where I open the modal; it is now fixed

File: src/components/Registration/RegistrationStepOne/RegistrationStepOne.tsx (hunk 2 above)
Comment (d-al-ibm) [reply src/components/Registration/RegistrationStepOne/RegistrationStepOne.tsx]: if you mean declare `useForm` inside `RegistrationStepOneRadioButtons` , I just tried and it does not work (useForm declaration must be single, if you declare multple times checks are not performed)
this structure comes from React Form Hook documentation about using it with dinamic components: https://react-hook-form.com/advanced-usage#ConditionalControlledComponent

**Pull request merged**